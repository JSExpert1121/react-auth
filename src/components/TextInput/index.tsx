// @material ui components
import TextField from '@material-ui/core/TextField';

// i18n
import authMsgs from 'translations/messages/page.auth';

// custom components
import withValidators from 'components/HOCs/withValidators';
import {
    isValidEmail,
    isRequired,
    isValidChar,
    isValidLengthMin,
    isValidLengthMax,
    isPasswordMatched
} from 'helper/validators';

export const EmailInput = withValidators([
    { validator: isRequired, message: authMsgs.emailRequired },
    { validator: isValidEmail, message: authMsgs.emailInvalid }
])(TextField);

export const NameInput = withValidators([
    { validator: isRequired, message: authMsgs.nameRequired }
])(TextField);

export const PassInput = withValidators([
    { validator: isRequired, message: authMsgs.passRequired },
    { validator: isValidChar, message: authMsgs.passInvalid },
])(TextField);

export const PassConfirmInput = withValidators([
    { validator: isRequired, message: authMsgs.passRequired },
    { validator: isValidChar, message: authMsgs.passInvalid },
    { validator: isValidLengthMin(8), message: authMsgs.passShort },
    { validator: isValidLengthMax(24), message: authMsgs.passLong },
    { validator: isPasswordMatched, message: authMsgs.passMismatch },
])(TextField);