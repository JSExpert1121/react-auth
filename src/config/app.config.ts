export const KAKAOMAP_API_KEY = '44a509cf4bf161e250c940454df5977c';
export const SUPPORTED_COUNTRIES = ['kr', 'jp', 'cn', 'hk', 'sg', 'tw', 'th', 'my', 'au', 'us', 'gb', 'de', 'fr', 'kz', 'ae', 'in'];
// export const COUNTRY_CODES = ['82', '81', '86', '852', '65', '886', '66', '60', '61', '1', '44', '49', '49', '33', '7', '971', '91'];
