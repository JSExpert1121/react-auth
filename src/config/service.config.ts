export const API_GATEWAY_ADDRESS = 'https://apig.t.moria.hitit.xyz:8765';
export const OAUTH_CLIENT_ID = 'web';
export const OAUTH_CLIENT_SECRET = 'hititweb42!';