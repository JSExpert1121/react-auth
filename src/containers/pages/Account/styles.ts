import { Theme, createStyles } from '@material-ui/core/styles';
import Colors from 'assets/colors';

export const styleFn = (theme: Theme) => createStyles({
    root: {
        width: '100%',
        padding: theme.spacing(3),
        maxWidth: 512,
        minWidth: 480,
    },
    form: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: '100%',
        textAlign: 'center'
    },
    logo: {
        padding: theme.spacing(2, 0)
    },
    title: {
        fontSize: '1.5rem',
        color: 'rgba(0, 0, 0, 0.8)',
        padding: theme.spacing(2)
    },
    text: {
        fontSize: '0.875rem',
        color: 'rgba(0, 0, 0, 0.7)',
        padding: theme.spacing(2)
    },
    success: {
        color: 'rgba(0, 0, 0, 0.9)',
        padding: theme.spacing(1)
    },
    error: {
        fontSize: '0.875rem',
        color: Colors.error,
        padding: theme.spacing(0, 0, 0, 2)
    },
    busy: {
        width: '100%',
    },
    submit: {
        margin: theme.spacing(2, 0),
        height: 'calc(100% - 12px)'
    },
    bottomMenu: {
        width: '100%',
        display: 'flex',
        padding: theme.spacing(2, 0)
    }
});