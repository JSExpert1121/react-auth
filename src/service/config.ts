import { oauthClient } from './base';

export const setLanguage = (lang: string) => {
    oauthClient.setLanguage(lang);
}
