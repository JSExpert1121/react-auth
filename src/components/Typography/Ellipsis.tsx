import React from 'react';
import { makeStyles, Theme } from '@material-ui/core/styles';

interface IEllipsisProps {
    fontSize?: string;
    maxLines?: number;
    hidden?: boolean;
}

const useStyles = (props: IEllipsisProps) => makeStyles((theme: Theme) => ({
    ellipsis: {
        display: '-webkit-box',
        maxHeight: `calc(${props.maxLines} * ${1.5} * ${props.fontSize})`,
        lineHeight: '1.5',
        visibility: !!props.hidden ? 'hidden' : 'visible',
        fontSize: props.fontSize,
        '-webkitLineClamp': props.maxLines,
        '-webkitBoxOrient': 'vertical',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
    }
}));

const EllipsisText: React.FC<IEllipsisProps> = (props) => {
    const { children, fontSize, maxLines, hidden } = props;
    let realSize = fontSize || '1rem';
    let realLines = maxLines || 2;

    const classes = useStyles({ fontSize: realSize, maxLines: realLines, hidden })({});
    return (
        <span className={classes.ellipsis}>
            {children}
        </span>
    );
};

export default EllipsisText;
