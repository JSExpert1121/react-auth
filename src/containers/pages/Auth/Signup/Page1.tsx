/**
 * @name SignupPage
 * @description Page for user registration. email/name/password/phone
 * @version 1.2
 * @todo API integration
 * @todo Terms and Conditions
 * @author Darko
 */
import * as React from 'react';
import { Link as RouterLink } from 'react-router-dom';

// material ui components & icons
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import LinearProgress from '@material-ui/core/LinearProgress';
import Avatar from '@material-ui/core/Avatar';
import CssBaseline from '@material-ui/core/CssBaseline';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import FormControl from '@material-ui/core/FormControl';
import IconButton from '@material-ui/core/IconButton';
import Link from '@material-ui/core/Link';
import { createStyles, Theme, withStyles, StyledComponentProps } from '@material-ui/core/styles';
import AccountCircleOutlined from '@material-ui/icons/AccountCircleOutlined';
import Visibility from '@material-ui/icons/Visibility';
import VisibilityOff from '@material-ui/icons/VisibilityOff';

// 3rd party libraries
import { MessageDescriptor, injectIntl, IntlShape } from 'react-intl';
import messages from 'translations/messages/page.auth';
import globalMsgs from 'translations/messages/global';

// custom components
import PhoneInput from 'containers/others/PhoneInput';

// actions & apis & types
import * as MemberApi from 'service/member';

// input elements with custom validations
import { EmailInput, NameInput, PassConfirmInput } from 'components/TextInput';


// component styles
const styles = (theme: Theme) => createStyles({
    root: {
        padding: theme.spacing(4, 2),
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: 512
    },
    form: {
        display: 'flex',
        width: '100%',
        flex: 1,
        flexDirection: 'column',
        alignItems: 'center',
        minHeight: 548
    },
    busy: {
        width: '100%',
        marginTop: theme.spacing(2)
    },
    avatar: {
        margin: theme.spacing(1),
        backgroundColor: theme.palette.primary.dark,
    },
    content: {
        width: '100%',
        textAlign: 'left',
        padding: theme.spacing(1),
        display: 'flex',
        flex: 1,
        flexDirection: 'column'
    },
    row: {
        paddingTop: theme.spacing(2),
    },
    countryCode: {
        marginTop: theme.spacing(1)
    },
    submit: {
        margin: theme.spacing(1, 0),
        minWidth: 120,
        height: 'calc(100% - 12px)'
    },
    getcode: {
        height: '100%'
    },
    link: {
        cursor: 'pointer'
    },
    modal: {
        width: '50%',
        height: '80%'
    }
});

// component props and states
export interface ISignupPage1Props extends StyledComponentProps {
    goNext: (email: string, name: string, password: string, confirm: string, countryCode: string, phoneNo: string, phoneVerified: boolean) => void;
    intl: IntlShape;
}

interface ISignupPage1State {
    isBusy: boolean;
    valid: boolean;
    email: string;
    emailMsg?: MessageDescriptor;
    name: string;
    nameMsg?: MessageDescriptor;
    password: string;
    passMsg?: MessageDescriptor;
    confirm: string;
    confirmMsg?: MessageDescriptor;
    countryCode: string;
    phoneNo: string;
    phoneVerified: boolean;
    emailAvailable: boolean;
    showPassword: boolean;
}

export class SignupPage1 extends React.Component<ISignupPage1Props, ISignupPage1State> {
    constructor(props: Readonly<ISignupPage1Props>) {
        super(props);

        this.state = {
            isBusy: false,
            valid: true,
            email: '',
            emailMsg: undefined,
            name: '',
            nameMsg: undefined,
            password: '',
            passMsg: undefined,
            confirm: '',
            confirmMsg: undefined,
            countryCode: '82',
            phoneNo: '',
            phoneVerified: false,
            emailAvailable: false,
            showPassword: false
        }
    }

    // email input event handler
    emailChange = (value: string, message: MessageDescriptor) => {
        this.setState({
            email: value,
            emailMsg: message
        });
    }

    emailBlur = async () => {
        if (this.state.emailMsg || this.state.email.length === 0) return;
        this.setState({ isBusy: true });
        try {
            await MemberApi.checkEmail(this.state.email);
            this.setState({
                isBusy: false,
                emailAvailable: true
            });
        } catch (error) {
            this.setState({
                emailMsg: messages.emailUsed,
                emailAvailable: false,
                isBusy: false
            });
        }
    }

    // name input event handler
    nameChange = (value: string, message: MessageDescriptor) => {
        this.setState({
            name: value,
            nameMsg: message
        });
    }

    // password input event handlers
    passChange = (value: string, message: MessageDescriptor) => {
        if (message) {
            this.setState({
                password: value,
                passMsg: message
            });
        } else {
            this.setState({
                password: value,
                passMsg: undefined,
                confirmMsg: undefined
            });
        }
    }

    confirmChange = (value: string, message: MessageDescriptor) => {
        if (message) {
            this.setState({
                confirm: value,
                confirmMsg: message
            });
        } else {
            this.setState({
                confirm: value,
                passMsg: undefined,
                confirmMsg: undefined
            });
        }
    }

    togglePasswordVisibility = () => {
        this.setState({
            showPassword: !this.state.showPassword
        })
    }

    // phone verified event handler(from child component)
    handlePhone = (country: string, phone: string, verified: boolean) => {
        this.setState({
            phoneNo: phone,
            phoneVerified: verified,
            countryCode: country
        });
    }

    // next page(TAC)
    next = () => {
        const { email, name, password, confirm, countryCode, phoneNo, phoneVerified } = this.state;
        this.props.goNext(email, name, password, confirm, countryCode, phoneNo, phoneVerified);
    }

    setBusy = (busy: boolean) => {
        this.setState({ isBusy: busy });
    }

    render() {
        const { classes, intl } = this.props;
        const { isBusy, emailMsg, email, name, nameMsg,
            password, passMsg, confirm, confirmMsg, countryCode,
            phoneNo, phoneVerified, emailAvailable, showPassword
        } = this.state;

        // enable signup button
        const joinable = emailAvailable && !nameMsg && !passMsg && !confirmMsg && phoneVerified && password.length > 0 && confirm.length > 0;

        return (
            <Paper square className={classes && classes.root}>
                <CssBaseline />
                <FormControl className={classes && classes.form}>
                    <Avatar className={classes && classes.avatar}>
                        <AccountCircleOutlined />
                    </Avatar>
                    <Typography component="h1" variant='h5'>
                        {intl.formatMessage(messages.signupCaption)}
                    </Typography>
                    <Box className={classes && classes.content}>
                        <LinearProgress id='busy-indicator' className={classes && classes.busy} style={{ visibility: isBusy ? 'visible' : 'hidden' }} />
                        <Box>
                            <EmailInput
                                id='email'
                                name='email'
                                value={email}
                                type='email'
                                autoFocus
                                label={intl.formatMessage(messages.emailLabel)}
                                helperText={intl.formatMessage(messages.emailHelper)}
                                handleChange={this.emailChange}
                                message={emailMsg}
                                onBlur={this.emailBlur}
                            />
                        </Box>
                        <Box className={classes && classes.row}>
                            <NameInput
                                id='name'
                                name='name'
                                value={name}
                                label={intl.formatMessage(messages.nameLabel)}
                                helperText={intl.formatMessage(messages.nameHelper)}
                                handleChange={this.nameChange}
                                message={nameMsg}
                            />
                        </Box>
                        <Box className={classes && classes.row} style={{ flex: 1 }}>
                            <Box style={{ display: 'flex' }}>
                                <Grid container spacing={1} style={{ flex: 1 }}>
                                    <Grid item xs={6}>
                                        <PassConfirmInput
                                            id="password"
                                            name="password"
                                            type={showPassword ? 'text' : 'password'}
                                            value={password}
                                            label={intl.formatMessage(messages.passLabel)}
                                            helperText={intl.formatMessage(messages.passHelper)}
                                            compare={confirm}
                                            handleChange={this.passChange}
                                            message={passMsg}
                                        />
                                    </Grid>
                                    <Grid item xs={6}>
                                        <PassConfirmInput
                                            id="confirm-password"
                                            name="password"
                                            type={showPassword ? 'text' : 'password'}
                                            value={confirm}
                                            label={intl.formatMessage(messages.confirmLabel)}
                                            handleChange={this.confirmChange}
                                            compare={password}
                                            message={confirmMsg}
                                        />
                                    </Grid>
                                </Grid>
                                <Box style={{ display: 'flex', flexDirection: 'column' }}>
                                    <IconButton
                                        edge="end"
                                        aria-label="password-visibility"
                                        name="passwordVisibility"
                                        id="passwordVisibility"
                                        style={{ paddingTop: 16 }}
                                        onClick={this.togglePasswordVisibility}
                                    >
                                        {showPassword ? <Visibility /> : <VisibilityOff />}
                                    </IconButton>
                                </Box>
                            </Box>
                        </Box>
                        <PhoneInput
                            setBusy={this.setBusy}
                            verify={this.handlePhone}
                            phoneNo={phoneNo}
                            countryCode={countryCode}
                        />
                        <Grid container className={classes.row}>
                            <Grid item xs={12}>
                                <Button
                                    id='next-button'
                                    type="submit"
                                    fullWidth
                                    variant="contained"
                                    color="primary"
                                    className={classes && classes.submit}
                                    disabled={!joinable}
                                    onClick={this.next}
                                >
                                    {intl.formatMessage(globalMsgs.next)}
                                </Button>
                            </Grid>
                            <Grid item xs={12} style={{ textAlign: 'center', paddingTop: 8 }}>
                                <Link to="/login" component={RouterLink}>
                                    {intl.formatMessage(messages.loginCaption)}
                                </Link>
                            </Grid>
                        </Grid>
                    </Box>
                </FormControl>
            </Paper >
        );
    }
}

export default injectIntl(withStyles(styles)(SignupPage1));