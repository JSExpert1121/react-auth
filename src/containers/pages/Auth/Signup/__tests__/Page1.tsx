/* disable tslint */
import React from 'react';
import { MemoryRouter, withRouter } from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import { createMount } from '@material-ui/core/test-utils';
import { ThemeProvider } from '@material-ui/styles';
import { IntlProvider } from "react-intl";
import { act } from 'react-dom/test-utils';

import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import AccountCircleOutlined from '@material-ui/icons/AccountCircleOutlined';
import IconButton from '@material-ui/core/IconButton';
import VisibilityOff from '@material-ui/icons/VisibilityOff';
import MuiPhoneInput from 'material-ui-phone-number';

import Page1, { SignupPage1 } from '../Page1';
import { EmailInput, NameInput, PassConfirmInput } from 'components/TextInput';

import { translationMessages } from 'config/i18n.config';
import theme from 'config/theme.config';
import messages from 'translations/messages/page.auth';

configure({ adapter: new Adapter() });
jest.mock('service/member');

describe('containers/pages/auth/signup/page1 UI components', () => {

    const goNext = jest.fn();
    const mount = createMount();

    const wrapper = mount(
        <MemoryRouter>
            <IntlProvider locale='en' messages={translationMessages.en}>
                <ThemeProvider theme={theme}>
                    <Page1 goNext={goNext} />
                </ThemeProvider>
            </IntlProvider>
        </MemoryRouter>
    );

    test('Snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('UI components', async (done) => {
        setImmediate(() => {
            const page = wrapper.find(SignupPage1);
            expect(page).toHaveLength(1);

            expect(page.find(Typography).at(0).props().children).toEqual('Create your Hit! T Account');
            expect(page.find(Avatar).at(0).props().children.type).toEqual(AccountCircleOutlined);

            expect(page.find(EmailInput)).toHaveLength(1);
            expect(page.find(EmailInput).at(0).props().value).toEqual('');
            expect(page.find(EmailInput).at(0).props().autoFocus).toEqual(true);
            expect(page.find(EmailInput).at(0).props().label).toEqual('Email Address');
            expect(page.find(EmailInput).at(0).props().helperText).toEqual('This email address will be used as your Hit! T account');
            expect(page.find(EmailInput).at(0).props().message).toEqual(undefined);

            expect(page.find(NameInput)).toHaveLength(1);
            expect(page.find(NameInput).at(0).props().value).toEqual('');
            expect(page.find(NameInput).at(0).props().label).toEqual('Name');
            expect(page.find(NameInput).at(0).props().message).toEqual(undefined);

            expect(page.find(PassConfirmInput)).toHaveLength(2);
            expect(page.find(PassConfirmInput).at(0).props().value).toEqual('');
            expect(page.find(PassConfirmInput).at(1).props().value).toEqual('');
            expect(page.find(PassConfirmInput).at(0).props().label).toEqual('Password');
            expect(page.find(PassConfirmInput).at(1).props().label).toEqual('Confirm password');
            expect(page.find(PassConfirmInput).at(0).props().type).toEqual('password');
            expect(page.find(PassConfirmInput).at(1).props().type).toEqual('password');
            expect(page.find(PassConfirmInput).at(0).props().compare).toEqual('');
            expect(page.find(PassConfirmInput).at(1).props().compare).toEqual('');
            expect(page.find(PassConfirmInput).at(0).props().message).toEqual(undefined);
            expect(page.find(PassConfirmInput).at(1).props().message).toEqual(undefined);

            expect(page.find(IconButton).at(0).props().children.type).toEqual(VisibilityOff);

            expect(page.find(MuiPhoneInput)).toHaveLength(1);
            const props = page.find(MuiPhoneInput).props() as any;
            expect(props.defaultCountry).toEqual('ko');
            expect(props.fullWidth).toEqual(true);
            expect(props.label).toEqual('Mobile Number');
            expect(props.disabled).toEqual(false);

            expect(page.find('#send-code-btn').first()).toHaveLength(1);
            expect(page.find('#send-code-btn').first().props().children).toEqual('Get Code');

            expect(page.find('#next-button').first()).toHaveLength(1);
            expect(page.find('#next-button').first().props().children).toEqual('NEXT');
            expect(page.find('#next-button').first().props().disabled).toEqual(true);

            done();
        });
    });
});

describe('containers/pages/auth/signup/page1 email interaction', () => {

    const goNext = jest.fn();
    const mount = createMount();

    const wrapper = mount(
        <MemoryRouter>
            <IntlProvider locale='en' messages={translationMessages.en}>
                <ThemeProvider theme={theme}>
                    < Page1 goNext={goNext} />
                </ThemeProvider>
            </IntlProvider>
        </MemoryRouter>
    );

    test('Email change event(invalid address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: '123@cc.a' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().message).toEqual(messages.emailInvalid);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email change event(empty string)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: '' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().message).toEqual(messages.emailRequired);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email blur event(existing address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('blur');
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().message).toEqual(messages.emailRequired);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email change event(valid address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: '123@cc.au' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email blur event(existing address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: 'user1@test.com' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().value).toEqual('user1@test.com');
            expect(page.find(EmailInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email blur event(existing address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('blur');
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().value).toEqual('user1@test.com');
            expect(page.find(EmailInput).first().props().message).toEqual(messages.emailUsed);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email blur event(valid address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: 'user4@test.com' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().value).toEqual('user4@test.com');
            expect(page.find(EmailInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email blur event(valid address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('blur');
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().value).toEqual('user4@test.com');
            expect(page.find(EmailInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });
});

describe('containers/pages/auth/signup/page1 name interaction', () => {
    const goNext = jest.fn();
    const mount = createMount();

    const wrapper = mount(
        <MemoryRouter>
            <IntlProvider locale='en' messages={translationMessages.en}>
                <ThemeProvider theme={theme}>
                    < Page1 goNext={goNext} />
                </ThemeProvider>
            </IntlProvider>
        </MemoryRouter>
    );


    test('Name change event(empty string)', done => {
        act(() => {
            wrapper.find('#name').last().simulate('change', { target: { value: '' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(NameInput).first().props().value).toEqual('');
            expect(page.find(NameInput).first().props().message).toEqual(messages.nameRequired);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Name change event(valid name)', done => {
        act(() => {
            wrapper.find('#name').last().simulate('change', { target: { value: 'user' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(NameInput).first().props().value).toEqual('user');
            expect(page.find(NameInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });
});

describe('containers/pages/auth/signup/page1 password interaction', () => {
    const goNext = jest.fn();
    const mount = createMount();

    const wrapper = mount(
        <MemoryRouter>
            <IntlProvider locale='en' messages={translationMessages.en}>
                <ThemeProvider theme={theme}>
                    < Page1 goNext={goNext} />
                </ThemeProvider>
            </IntlProvider>
        </MemoryRouter>
    );


    test('Password and confirm change event(empty string)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '' } });
            wrapper.find('#confirm-password').last().simulate('change', { target: { value: '' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).first().props().value).toEqual('');
            expect(page.find(PassConfirmInput).first().props().message).toEqual(messages.passRequired);
            expect(page.find(PassConfirmInput).last().props().value).toEqual('');
            expect(page.find(PassConfirmInput).last().props().message).toEqual(messages.passRequired);
            done();
        });
    });

    test('Password and confirm change event(less than 8 chars)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '1232345' } });
            wrapper.find('#confirm-password').last().simulate('change', { target: { value: '2232345' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).first().props().value).toEqual('1232345');
            expect(page.find(PassConfirmInput).first().props().message).toEqual(messages.passShort);
            expect(page.find(PassConfirmInput).last().props().value).toEqual('2232345');
            expect(page.find(PassConfirmInput).last().props().message).toEqual(messages.passShort);
            done();
        });
    });

    test('Password change event(more than 24 chars)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '1223232323232323323233233232345' } });
            wrapper.find('#confirm-password').last().simulate('change', { target: { value: '2223232323232323323233233232345' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).first().props().value).toEqual('1223232323232323323233233232345');
            expect(page.find(PassConfirmInput).first().props().message).toEqual(messages.passLong);
            expect(page.find(PassConfirmInput).last().props().value).toEqual('2223232323232323323233233232345');
            expect(page.find(PassConfirmInput).last().props().message).toEqual(messages.passLong);
            done();
        });
    });

    test('Password change event(invalid chars)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '32332323°3233232345' } });
            wrapper.find('#confirm-password').last().simulate('change', { target: { value: '42332323°3233232345' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).first().props().value).toEqual('32332323°3233232345');
            expect(page.find(PassConfirmInput).first().props().message).toEqual(messages.passInvalid);
            expect(page.find(PassConfirmInput).last().props().value).toEqual('42332323°3233232345');
            expect(page.find(PassConfirmInput).last().props().message).toEqual(messages.passInvalid);
            done();
        });
    });

    test('Password change event(invalid chars)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '323323233233232345' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).first().props().value).toEqual('323323233233232345');
            expect(page.find(PassConfirmInput).first().props().message).toEqual(messages.passMismatch);
            done();
        });
    });

    test('Password change event(valid passwords)', done => {
        act(() => {
            wrapper.find('#confirm-password').last().simulate('change', { target: { value: '323323233233232345' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).last().props().value).toEqual('323323233233232345');
            expect(page.find(PassConfirmInput).first().props().message).toEqual(undefined);
            expect(page.find(PassConfirmInput).last().props().message).toEqual(undefined);
            done();
        });
    });

    test('toggle password show', done => {
        act(() => {
            wrapper.find('#passwordVisibility').last().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).last().props().type).toEqual('text');
            expect(page.find(PassConfirmInput).first().props().type).toEqual('text');
            done();
        });
    });

    test('toggle password hide', done => {
        act(() => {
            wrapper.find('#passwordVisibility').last().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).last().props().type).toEqual('password');
            expect(page.find(PassConfirmInput).first().props().type).toEqual('password');
            done();
        });
    });
});

describe('containers/pages/auth/signup/page1 signup action', () => {

    const goNext = jest.fn();
    const mount = createMount();

    const wrapper = mount(
        <MemoryRouter>
            <IntlProvider locale='en' messages={translationMessages.en}>
                <ThemeProvider theme={theme}>
                    < Page1 goNext={goNext} />
                </ThemeProvider>
            </IntlProvider>
        </MemoryRouter>
    );


    test('UI components', done => {

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find('#next-button').first().props().disabled).toBe(true);
            done();
        });
    });

    test('Email change event(valid address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: 'user4@test.com' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().value).toEqual('user4@test.com');
            expect(page.find(EmailInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Email blur event(valid address)', done => {
        act(() => {
            wrapper.find('#email').last().simulate('blur');
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(EmailInput).first().props().value).toEqual('user4@test.com');
            expect(page.find(EmailInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Name change event(valid name)', done => {
        act(() => {
            wrapper.find('#name').last().simulate('change', { target: { value: 'testuser4' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(NameInput).first().props().value).toEqual('testuser4');
            expect(page.find(NameInput).first().props().message).toEqual(undefined);
            expect(page.find('#send-code-btn').first().props().disabled).toEqual(true);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('Password change event(valid passwords)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '323323233233232345' } });
            wrapper.find('#confirm-password').last().simulate('change', { target: { value: '323323233233232345' } });
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find(PassConfirmInput).first().props().value).toEqual('323323233233232345');
            expect(page.find(PassConfirmInput).last().props().value).toEqual('323323233233232345');
            expect(page.find(PassConfirmInput).first().props().message).toEqual(undefined);
            expect(page.find(PassConfirmInput).last().props().message).toEqual(undefined);
            expect(page.find('#next-button').first().props().disabled).toEqual(true);
            done();
        });
    });

    test('phone change:', done => {
        act(() => {
            const page = wrapper.find(SignupPage1);
            page.instance().handlePhone('82', '1022938283', true);
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(page.find('#next-button').first().props().disabled).toEqual(false);
            done();
        });
    });

    test('next button handler', done => {
        act(() => {
            wrapper.find('#next-button').last().simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            const page = wrapper.find(SignupPage1);
            expect(goNext).toHaveBeenCalledWith(
                'user4@test.com',
                'testuser4',
                '323323233233232345',
                '323323233233232345',
                '82', '1022938283',
                true
            );
            done();
        });
    });

    test('setBusy function', done => {
        act(() => {
            wrapper.find(SignupPage1).instance().setBusy(false);
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(SignupPage1).state().isBusy).toBe(false);
            done();
        })
    })
});