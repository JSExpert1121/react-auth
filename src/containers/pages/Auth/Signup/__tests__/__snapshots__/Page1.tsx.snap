// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/auth/signup/page1 UI components Snapshot test 1`] = `ReactWrapper {}`;
